libyul/backends/wasm/EVMToEwasmTranslator.cpp

```diff
@@ -985,19 +1000,24 @@ function mload(x1, x2, x3, x4) -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(to_internal_i32ptr(x1, x2, x3, x4))
 }
 function mload_internal(pos:i32) -> z1, z2, z3, z4 {
-	z1 := endian_swap(i64.load(pos))
-	z2 := endian_swap(i64.load(i32.add(pos, 8:i32)))
-	z3 := endian_swap(i64.load(i32.add(pos, 16:i32)))
-	z4 := endian_swap(i64.load(i32.add(pos, 24:i32)))
+	z1 := bswap64(i64.load(pos))
+	z2 := bswap64(i64.load(i32.add(pos, 8:i32)))
+	z3 := bswap64(i64.load(i32.add(pos, 16:i32)))
+	z4 := bswap64(i64.load(i32.add(pos, 24:i32)))
+}
+function mload_address(pos:i32) -> z2, z3, z4 {
+	z2 := i64.and(bswap64(i64.load(pos)), 0x00000000ffffffff)
```

Comment:
Why not just read 32-bits here? (And adjust the offset in the caller too)

```suggestion
	z2 := bswa32(i32.load(pos))
```

Reply:
This would only work, if we do something like ` bswap32(i64.extend_i32_u(i32.load(i32.add(pos, 4:i32))))`. I think `i64.and(bswap64(i64.load(pos)), 0x00000000ffffffff)` is better readable.

Reply:
I don't really get it. As long as long you only put in 20 bytes of the address in the memory and `pos` refers to that offset there is no extension needed. Why would `bswap32` need `i64` inputs?

Reply:
`bswap32` is just defined like this - `function bswap32(x) -> y`. Maybe this can just be changed, I didn't tried that.

Reply:
I just changed that. I think it is better now.